### ISSUE TITLE
SDS removal broken in delta xds

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#49852]
Not sure if this is a real issue, opening to not forget

```
2024-03-11T21:33:54.648073Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/delta_subscription_state.cc:269  delta config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate() thread=25
2024-03-11T21:33:54.648674Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138    gRPC config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate()  thread=25
2024-03-11T21:33:54.648695Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138    gRPC config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate()  thread=25
2024-03-11T21:33:54.648706Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138    gRPC config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate()  thread=25
2024-03-11T21:33:54.648717Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138    gRPC config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate()  thread=25
2024-03-11T21:33:54.648728Z     warning envoy config external/envoy/source/extensions/config_subscription/grpc/grpc_subscription_impl.cc:138    gRPC config for type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret rejected: Missing SDS resources for kubernetes://ingress-credential in onConfigUpdate()  thread=25

```

**Comment by howardjohn** [comment]
https://github.com/envoyproxy/envoy/issues/32832

**Comment by keithmattix** [comment]
This was fixed in envoyproxy/envoy#33362
